File: 1.js
//@ts-ignore
const fs = require("fs");
const testInput = [199, 200, 208, 210, 200, 207, 240, 269, 260, 263];

const simpleIncreases = (data) => {
  const increases = data.reduce((acc, curr, i, arr) => {
    if (!arr[i + 1]) return acc;
    if (curr < arr[i + 1]) return acc + 1;
    return acc;
  }, 0);
  console.log(increases);
};

const sum = (array) => array.reduce((prev, curr) => prev + curr);

const slidingWindowIncreases = (data, windowSize) => {
  const increases = data.reduce((acc, curr, i, arr) => {
    if (arr.slice(i + 1, i + 1 + windowSize).length < 3) return acc;

    const slice1 = arr.slice(i, i + windowSize);
    const slice2 = arr.slice(i + 1, i + 1 + windowSize);

    if (sum(slice1) < sum(slice2)) return acc + 1;
    return acc;
  }, 0);
  console.log(increases);
};

try {
  const data = fs
    .readFileSync("inputs/1.txt", "utf8")
    .split("\r\n")
    .map((i) => parseInt(i, 10));
  // simpleIncreases(data);
  slidingWindowIncreases(data, 3);
} catch (err) {
  console.error(err);
}

File: 1.py
from functools import reduce

testData = [199, 200, 208, 210, 200, 207, 240, 269, 260, 263]

with open('inputs/1.txt', encoding="utf-8") as f:
    inputData = list(map(int, f.read().split()))
    
def compareData(x, y):
    if x < y:
        return 1
    return 0

def simpleIncrease(inputData):
    acc = 0
    for index, this in enumerate(inputData):
        try:
           acc += compareData(this, inputData[index+1])
        except IndexError:
            print("done: ", acc)

def slidingWindowIncrease(inputData, windowSize):
    acc = 0
    for index, this in enumerate(inputData):
        # sum of n=0 + windowSize, compare with sum of n+1 + windowSize
        try:
            slice1 = reduce(lambda x, y: x+y, inputData[index:index+windowSize])
            slice2 = reduce(lambda x, y: x+y, inputData[index+1:index+windowSize+1])
            acc += compareData(slice1, slice2)
            
        except (IndexError, TypeError):
            print("done: ", acc)

# simpleIncrease(inputData)
slidingWindowIncrease(inputData, 3)
